### ISSUE TITLE
Wrong ':' token evaluation in arrays

### ISSUE THREAD

**Issue by emaincourt** [babel/babel-eslint#512]
**Configuration**

* **ESLint Version: 3.19.0**
* **Node Version: 6.10**
* **npm Version: 3.10.10**

```js
module.exports = {
  parser: 'babel-eslint',
  extends: 'airbnb',
  env: {
    node: true,
    jest: true,
    jasmine: true,
  },
  rules: {
    'no-console': 'error',
    strict: 'off',
    'import/prefer-default-export': 'off',
    'jsx-a11y/anchor-is-valid': 'off',
    'jsx-a11y/href-no-hash': 'off',
  },
};
```

**Code**

```js
const foo = 'foo' : 'bar'; // Error raised
const bar = [
  'foo' : 'bar', // No error raised
];
```

I was expecting the `bar` declaration to raise an error since `'foo' : 'bar'` doesn't make, as far as I know, any sense outside of a ternary expression scope. But it actually led to an inability for babel to transpile the code. Is it an expected behaviour ? And if yes, what is it's purpose ?

**Comment by kaicataldo** [comment]
Thank you for the issue. Now that [`@babel/eslint-parser` has been released](https://babeljs.io/blog/2020/07/13/the-state-of-babel-eslint), we are making this repository read-only. If this is a change you would still like to advocate for, please reopen this in the [babel/babel monorepo](https://github.com/babel/babel).
